libs/designer/src/lib/ui/panel/configureTemplatePanel/configureWorkflowsPanel/tabs/selectWorkflowsTab.tsx

```diff
@@ -24,82 +46,219 @@ export const SelectWorkflows = () => {
+            {isLoading
+              ? logicAppName
+                ? [...Array(5)].map((_, index) => (
+                    <TableRow key={index} aria-hidden="true">
+                      <TableSelectionCell checkboxIndicator={{ 'aria-label': 'Loading row' }} checked={false} />
+                      <TableCell>
+                        <Skeleton aria-label="Loading name">
+                          <SkeletonItem />
+                        </Skeleton>
+                      </TableCell>
+                      <TableCell>
+                        <Skeleton aria-label="Loading trigger">
+                          <SkeletonItem />
+                        </Skeleton>
+                      </TableCell>
+                    </TableRow>
+                  ))
+                : null
+              : rows.map(({ item, selected, onClick, onKeyDown, appearance }) => (
+                  <TableRow key={item.id} onClick={onClick} onKeyDown={onKeyDown} aria-selected={selected} appearance={appearance}>
+                    <TableSelectionCell checked={isConsumption || selected} checkboxIndicator={{ 'aria-label': 'Select row' }} />
+                    <TableCell>
+                      <TableCellLayout>{item.name}</TableCellLayout>
+                    </TableCell>
+                    <TableCell>
+                      <TableCellLayout>{item.trigger}</TableCellLayout>
+                    </TableCell>
+                  </TableRow>
+                ))}
+          </TableBody>
+        </Table>
+      </TemplatesSection>
     </div>
   );
 };
 
+const normalizedWorkflowId = (workflowId: string) => workflowId.toLowerCase();
+
 export const selectWorkflowsTab = (
   intl: IntlShape,
   dispatch: AppDispatch,
-  { hasError, isSaving, onClosePanel }: ConfigureWorkflowsTabProps
+  {
+    isSaving,
+    isPrimaryButtonDisabled,
+    selectedWorkflowsList,
+    onWorkflowsSelected,
+    onNextButtonClick,
+  }: ConfigureWorkflowsTabProps & {
+    onWorkflowsSelected: (normalizedWorkflowIds: string[]) => void;
+    onNextButtonClick: () => Promise<void>;
+  }
 ): TemplateTabProps => ({
   id: constants.CONFIGURE_TEMPLATE_WIZARD_TAB_NAMES.SELECT_WORKFLOWS,
   title: intl.formatMessage({
     defaultMessage: 'Select workflows',
     id: 'vWOWFo',
     description: 'The tab label for the monitoring select workflows tab on the configure template wizard',
   }),
-  hasError: hasError,
-  content: <SelectWorkflows />,
+  content: <SelectWorkflows selectedWorkflowsList={selectedWorkflowsList} onWorkflowsSelected={onWorkflowsSelected} />,
   footerContent: {
     primaryButtonText: intl.formatMessage({
       defaultMessage: 'Next',
       id: '0UfxUM',
       description: 'Button text for moving to the next tab in the create workflow panel',
     }),
-    primaryButtonOnClick: () => {
+    primaryButtonOnClick: async () => {
       dispatch(selectPanelTab(constants.CONFIGURE_TEMPLATE_WIZARD_TAB_NAMES.CUSTOMIZE_WORKFLOWS));
+      await onNextButtonClick();
```

Comment:
There is no need to do await here, since you are not waiting on anything but moving to next tab